src/pudl/settings.py

```diff
@@ -269,7 +269,6 @@ def check_eia_dependencies(cls, values):  # noqa: N805
         """Make sure the dependencies between the eia datasets are satisfied.
 
         Dependencies:
-        * eia860 requires eia923.boiler_fuel_eia923 and eia923.generation_eia923.
```

Comment:
Tables aren't part of the settings anymore.

I also noticed that we do a lot of cartwheels in the BGA function to ensure that we're only ever operating on data where we have the same set of EIA-860 and EIA-923 years. Is that something we might want to enforce with the settings? Or does the EIA-860M mess that up?

Reply:
This feels oos for this pr but it is a good thing to consider ripping out. 860m shouldn't muck with the bga too much because it doesn't contain the bga table (860m only has the generator table)

Reply:
I don't understand. What would be good to consider ripping out? The `check_eia_dependencies` entirely?  The cartwheels that ensure only the same years of EIA-860 and EIA-923 are used in the BGA?

Are you just saying that we don't have to worry about the EIA-860M in this because the logic that depends on the `eia860m` parameter deals with it?

Reply:
i'm sorry i got turned around on what you were talking about. The `_restrict_years` cartwheeling in the bga process basically narrows  the options for what is assn-able to years that come from both 860 and 923. I guess we _could_ remove this if we always ensure in the settings that the 923 and 860 years are the same. we'd have to remove 860m data from the generators table to keep the outcome consistent.

but at a broader level i'm not sure we want to only allow 860 and 923 to be processed if the years are the same. `check_eia_dependencies` ensures that one dataset exists if you are trying to load another. i think it should be fine to load different years if say 860 is published before 923.

Reply:
Removed a bunch of unnecessary checking and consolidated the `_restrict_years` in the loop as you suggested.